src/app/view/view-controls/view-controls.component.ts

```diff
@@ -169,18 +177,21 @@ export class ViewControlsComponent implements OnInit, OnChanges, OnDestroy {
       return;
     }
 
-    const queryChanged = !areQueriesEqual(this.view.query, this.currentQuery);
-    const configChanged =
-      JSON.stringify(this.currentConfig[this.currentPerspective]) !==
-      JSON.stringify(this.view.config[this.currentPerspective]);
-
-    if (queryChanged || configChanged) {
+    if (this.queryChanged() || this.configChanged()) {
       this.askToDiscardChanges();
     } else {
       this.navigateToUrlWithoutView({});
     }
   }
 
+  private queryChanged(): boolean {
+    return !areQueriesEqual(this.view.query, this.currentQuery);
+  }
+
+  private configChanged(): boolean {
+    return JSON.stringify(this.currentConfig) !== JSON.stringify(this.view.config[this.currentPerspective]);
```

Comment:
This condition does not work for table perspective. We already have a selector for such comparison which is more sophisticated.